src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1371,29 +1373,49 @@ private ArrayTypeSymbol VisitArrayInitializer(BoundArrayCreation node)
                     if (hadNullabilityMismatch)
                     {
                         ReportDiagnostic(ErrorCode.WRN_NoBestNullabilityArrayElements, node.Syntax);
-                        checkConversions = false;
+                        checkNestedNullability = false;
                     }
                 }
-                if ((object)bestType == null)
+
+                if (bestType is null)
                 {
                     elementType = elementType.SetUnknownNullabilityForReferenceTypes();
-                    checkConversions = false;
+                    checkNestedNullability = false;
                 }
                 else
                 {
-                    elementType = TypeSymbolWithAnnotations.Create(bestType, BestTypeInferrer.GetNullableAnnotation(bestType, resultBuilder));
+                    elementType = TypeSymbolWithAnnotations.Create(bestType);
                 }
                 arrayType = arrayType.WithElementType(elementType);
             }
 
-            if (checkConversions && !elementType.IsValueType)
+            if (checkNestedNullability && !elementType.IsValueType)
             {
+                // Convert elements to best type to determine element top-level nullability and to report nested nullability warnings
                 for (int i = 0; i < n; i++)
                 {
                     var conversion = conversionBuilder[i];
                     var element = elementBuilder[i];
                     var resultType = resultBuilder[i];
-                    ApplyConversion(element, element, conversion, elementType, resultType, checkConversion: true, fromExplicitCast: false, useLegacyWarnings: false, AssignmentKind.Assignment);
+                    var obliviousBestType = elementType;
```

Comment:
>var obliviousBestType = elementType; [](start = 20, length = 36)

It looks like the ```elementType``` isn't necessary an oblivious type if the type is not being inferred.